qiskit_optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -222,7 +222,7 @@ def solve(self, problem: QuadraticProgram) -> MinimumEigenOptimizationResult:
         problem_ = self._convert(problem, self._converters)
 
         # construct operator and offset
-        operator, offset = problem_.to_ising()
+        operator, offset = problem_.to_ising(opflow=True)
```

Comment:
Where we use conversion methods internally, like here, and consume the operators ourselves, in this case via the min eigen solver, we could already move over to using SparsePauliOp now couldn't we. Is there any reason not to? Its what we would recommend right that people start the migration away from PauliSumOp (opflow) - ideally any new code would start out that way though of course its not the default yet for compatibility/migration reasons.

Reply:
Unfortunately, we cannot move to `SparsePauliOp` here because `MinimumEigenOptimizer` need to accept the legacy minimum eigensolvers that can deal with only opflow.

Reply:
Ah right, so it has to stay this way, even if the default of the to_ising gets changed, until the support for the legacy algorithms is removed.